diff --git a/Sources/SwiftTerm/Terminal.swift b/Sources/SwiftTerm/Terminal.swift
index ed08c28..9cf3e11 100644
--- a/Sources/SwiftTerm/Terminal.swift
+++ b/Sources/SwiftTerm/Terminal.swift
@@ -2307,5 +2307,6 @@ open class Terminal {
     func cmdEraseInLine (_ pars: [Int], _ collect: cstring)
     {
         let p = pars.count == 0 ? 0 : pars [0]
+        clampPendingWrapColumn()
        let leftBound = marginMode ? buffer.marginLeft : 0
        let rightBound = marginMode ? buffer.marginRight : cols - 1
@@ -2317,10 +2318,10 @@ open class Terminal {
         switch p {
         case 0:
-            eraseInBufferLine (y: buffer.y, start: buffer.x, end: rightBound + 1)
+            eraseInBufferLine (y: buffer.y, start: buffer.x, end: rightBound + 1, clearWrap: buffer.x == leftBound)
         case 1:
-            eraseInBufferLine (y: buffer.y, start: leftBound, end: buffer.x + 1)
+            eraseInBufferLine (y: buffer.y, start: leftBound, end: buffer.x + 1, clearWrap: buffer.x >= rightBound)
         case 2:
-            eraseInBufferLine (y: buffer.y, start: leftBound, end: rightBound + 1)
+            eraseInBufferLine (y: buffer.y, start: leftBound, end: rightBound + 1, clearWrap: true)
         default:
             break
         }
